spec/index.bs

```diff
@@ -3545,9 +3666,85 @@ interface mixin GPUProgrammablePassEncoder {
 };
 </script>
 
+{{GPUProgrammablePassEncoder}} has the following internal slots:
+
+<dl dfn-type=attribute dfn-for="GPUProgrammablePassEncoder">
+    : <dfn>\[[debug_group_stack]]</dfn> of type `sequence<DOMString>`.
+    ::
+        A stack of active debug group labels.
+</dl>
+
+## Debug Markers ## {#programmable-passes-debug-markers}
+
+Similar to Command encoders, programmable pass encoders provide methods to apply debug labels to
+groups of commands or insert a single label into the command sequence. Debug groups can be nested to
+create a hierarchy of labeled commands. These labels MAY be passed to the native API backends for
+tooling, MAY be used by the user agent's internal tooling, or MAY be a no-op when such tooling is
+not available or applicable.
+
 Debug groups in a {{GPUCommandEncoder}} or {{GPUProgrammablePassEncoder}}
 must be well nested.
 
+### <dfn method for=GPUProgrammablePassEncoder>pushDebugGroup(groupLabel)</dfn> ### {#GPUProgrammablePassEncoder-pushDebugGroup}
+
+<div algorithm="GPUProgrammablePassEncoder.pushDebugGroup">
+    <strong>|this|:</strong> of type {{GPUProgrammablePassEncoder}}.
+
+    **Arguments:**
+      - {{DOMString}} |groupLabel|
+
+    **Returns:** void
+
+    Marks the beginning of a labeled group of commands for the {{GPUProgrammablePassEncoder}}.
+
+    |groupLabel| defines the label for the command group.
+
+    On the [=Device timeline=], the following steps occur:
+
+        - If the user agent can fulfill the request:
+
+            - push |groupLabel| onto then end of |this|.{{GPUProgrammablePassEncoder/[[debug_group_stack]]}}.
+</div>
+
+### <dfn method for=GPUProgrammablePassEncoder>popDebugGroup()</dfn> ### {#GPUProgrammablePassEncoder-popDebugGroup}
+
+<div algorithm="GPUProgrammablePassEncoder.popDebugGroup">
+    <strong>|this|:</strong> of type {{GPUProgrammablePassEncoder}}.
+
+    **Returns:** void
+
+    Marks the end of a labeled group of commands for the {{GPUProgrammablePassEncoder}}.
+
+    On the [=Device timeline=], the following steps occur:
+
+        - If the user agent can fulfill the request and
```

Comment:
what happens if the request can not be fulfilled?

Reply:
Borrowed that verbiage from elsewhere in the doc, but after thinking it through I don't think it makes much sense. The push/pop should always happen if the call is valid.